src/app/file/file.cpp

```diff
@@ -529,6 +530,10 @@ FileOp* FileOp::createSaveDocumentOperation(const Context* context,
     Sprite* spr = fop->m_document->sprite();
     frame_t outputFrame = 0;
 
+    std::string newfn = base::get_file_name(fn);
+    base::replace_string(newfn, "{title}", base::get_file_title(fop->m_document->name()));
```

Comment:
The `filename_formatter()` function (called below) already should do this, so the source of this problem is from the Export command (`SaveFileCopyAsCommand::onExecute`) which calls `createSaveDocumentOperation()` with `filename`=the string that the user inserted in the output filename. Maybe, when the user put some text like `{title}` (or any other {...}) the filename should be the original sprite name, and its `SaveFileBaseCommand::m_filenameFormat` should be equal to the inserted text, which now specifies a format, and not the final filename).